File: Golang/350. Intersection of Two Arrays II.go
func intersect(nums1 []int, nums2 []int) []int {
    dict1 := make(map[int]int, len(nums1))
    dict2 := make(map[int]int, len(nums2))
    for _, num := range nums1 {
        dict1[num]++
    }
    for _, num := range nums2 {
        dict2[num]++
    }
    ans := []int{}
    for num := range dict1 {
        j := min(dict1[num], dict2[num])
        for k := 0; k < j; k++ {
            ans = append(ans, num)
        }
    }
    return ans
}

func min(a, b int) int {
    if a > b {
        return b
    } else {
        return a
    }
}

File: Python3/350. Intersection of Two Arrays II.py
class Solution:
    def intersect(self, nums1, nums2):
        """
        :type nums1: List[int]
        :type nums2: List[int]
        :rtype: List[int]
        """
        nums1 = collections.Counter(nums1)
        nums2 = collections.Counter(nums2)
        ans = []
        for num1 in nums1:
            if num1 in nums2:
                ans.extend([num1] * min(nums1[num1], nums2[num1]))
        return ans
